### ISSUE TITLE
vimperator: missing bash shorcuts

### ISSUE THREAD

**Issue by pale3** [vimperator/vimperator-labs#448]
There are missing bash shourcuts when input mode is active and while cursor is in textarea. Those shurtcuts works well when normal mode is on, but not in insert mode. Missing ones are 
```
c-w # kill word backword (essential)
a-backspace # basically same as above
s-insert # paste buffer (essential)
a-d # kill word ahead
c-a # go to begiining of line
c-e # go to end of line
```
Not every shorcut is important, at least those which I marked
as essential are of every day use.

**Comment by timss** [comment]
Not seeing all the same. Can you please specify OS, Firefox and Vimperator version?

Kubuntu 15.10, FF 44.0.2, Vimperator 3.12.0 – fresh profile

**Working**:

* `<C-w>`
* `<S-Insert>` – this is usually dictated by the system, not your browser
* `<C-a>`
* `<C-e>`

**Missing:**

* `<A-BS>` – also know as `<M-BS>`
* `<A-d>`

Relevant: #101

**Comment by pale3** [comment]
`c-w`
Main culrpit here was addon CustomizeShortcut, which I used only to disable accidental close of ff. Using this addon I disabled `c-s-w`.
Mapping `c-s-w` to <Nop> in insert mode doesnt stop accidental close 
above!

`S-Insert` works, but in the way that it only paste clipboard content, not buffer content which I want!  

OS: Archlinux
FF: 44
Vimperator: 3.12.0

**Comment by pale3** [comment]
I dont get why
map `<c-s-w> <Nop>` doesnt respect all modes. 

To solve accidental close of ff I need to map multiple `<c-s-w>` for every mode seperatly, like this:

```
cmap <c-s-w> <Nop>
imap <c-s-w> <Nop>
nmap <c-s-w> <Nop>
```

**Comment by timss** [comment]
>`S-Insert` works, but in the way that it only paste clipboard content, not buffer content which I want!

Are you certain this is not indeed the default behavior of Firefox without Vimperator? I believe these keybindings are dictated by X11, not Vimperator or Firefox for that matter. Moreover, they are platform dependant so it is difficult to include workarounds in an addon such as Vimperator. 

>To solve accidental close of ff I need to map multiple <c-s-w> for every mode seperatly

That is the intended behavior, just as described in the help pages for `map` (`:h map`):

>Below is an overview of which modes each map command applies to:
>
>:map   :noremap   :unmap   :mapclear  – both Normal and Visual modes
>:nmap  :nnoremap  :nunmap  :nmapclear – Normal mode
>:vmap  :vnoremap  :vunmap  :vmapclear – Visual mode
>:imap  :inoremap  :iunmap  :imapclear – Insert mode
>:cmap  :cnoremap  :cunmap  :cmapclear – Command-line mode

**Comment by timss** [comment]
I'm not sure if adding `<A-BS>` and `<A-d>` is such a good idea either because how messy `<A-*>` or `<M-*>` can be between different platforms (e.g. OS X vs Linux). What do you think @gkatsev ?

**Comment by pale3** [comment]
> Are you certain this is not indeed the default behavior of Firefox without Vimperator? I believe these keybindings are dictated by X11, not Vimperator or Firefox for that matter. Moreover, they are platform dependant so it is difficult to include workarounds in an addon such as Vimperator. 

Yes, this is defualt firefox behavior. In firefox `<S-Insert>` is mapped to 
clipboard paste. There were feature request and even patches for this, but somehow they didn't get merged. 

Older vimperator version have this feature. Here is info that feature is
removed starting from vimperator 3.9.1. 

>We don't override <S-Insert> in textboxes anymore, so it does the default
action (which is pasting the clipboard)

May we revert this behavior to be optional?

**Comment by timss** [comment]
There's some more information here: #169 

I suppose there could be an option which enables or disables the `<S-Insert>` map to use the now removed `pasteClipboard` function. A pull request would be most welcome. 

A temporary workaround could be to incorporate the previous behavior into your RC file using `:javascript <<EOF` (see `:h js`).

**Comment by pale3** [comment]
I made a new optional usage of pasteClipboard function in PR #452. I hope this will be merged!

**Comment by timss** [comment]
Cool, thanks!

I'll leave this up to one of the other maintainers to merge.
